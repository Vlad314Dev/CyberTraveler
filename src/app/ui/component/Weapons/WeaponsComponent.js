import './WeaponsStyle';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import {
    DEFAULT_WEAPON_TYPE,
    MISSILE_WEAPON_TYPE,
    PIERCE_WEAPON_TYPE 
} from 'UIStore/Weapon/WeaponConfig';

class WeaponsComponent extends PureComponent
{
    static propTypes = {
        selectedWeapon: PropTypes.string.isRequired,
        availableWeapons: PropTypes.array.isRequired
    };

    isWeaponAvailable(weaponType)
    {
        const { 
            availableWeapons 
        } = this.props;

        return availableWeapons.some((weapon) => weapon == weaponType);
    }

    render()
    {
        const {
            selectedWeapon
        } = this.props;

        return (
            <div block="Weapon" elem="Wrapper">
                <div block="Weapon" elem="List">
                    <div block="Weapon" mods={ { type: 'default', isSelected: selectedWeapon == 'default' } }>
                        <span block="Weapon" elem="Availability" mods={ { isAvailable: this.isWeaponAvailable(DEFAULT_WEAPON_TYPE) } }></span>
                    </div>
                    <div block="Weapon" mods={ { type: 'pierce', isSelected: selectedWeapon == 'pierce' } }>
                        <span block="Weapon" elem="Availability" mods={ { isAvailable: this.isWeaponAvailable(PIERCE_WEAPON_TYPE) } }></span>
                    </div>
                    <div block="Weapon" mods={ { type: 'missile', isSelected: selectedWeapon == 'missile' } }>
                        <span block="Weapon" elem="Availability" mods={ { isAvailable: this.isWeaponAvailable(MISSILE_WEAPON_TYPE) } }></span>
                    </div>
                </div>
            </div>
        );
    }
}

export default WeaponsComponent;
