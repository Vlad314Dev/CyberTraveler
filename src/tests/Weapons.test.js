/* eslint-disable no-undef */
import { shallow } from 'enzyme';
import React from 'react';

import WeaponsComponent from '../app/ui/component/Weapons/WeaponsComponent';

describe('MainMenuTitle', () => {
    it('Should render', () => {
        // eslint-disable-next-line no-unused-vars
        const component = shallow(
            <WeaponsComponent 
                selectedWeapon={ '' }
                availableWeapons={ [] }
            />
        );
    });

    it('Should render layout', () => {
        const component = shallow(
            <WeaponsComponent 
                selectedWeapon={ '' }
                availableWeapons={ [] }
            />
        );

        expect(component.getElements()).toMatchSnapshot();
    });
});
