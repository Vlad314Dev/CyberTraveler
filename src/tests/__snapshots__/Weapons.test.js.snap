// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenuTitle Should render layout 1`] = `
Array [
  <div
    className="Weapon__Wrapper"
  >
    <div
      className="Weapon__List"
    >
      <div
        className="Weapon Weapon_type_default"
      >
        <span
          className="Weapon__Availability"
        />
      </div>
      <div
        className="Weapon Weapon_type_pierce"
      >
        <span
          className="Weapon__Availability"
        />
      </div>
      <div
        className="Weapon Weapon_type_missile"
      >
        <span
          className="Weapon__Availability"
        />
      </div>
    </div>
  </div>,
]
`;
